07-contributor-guide.Rmd

```diff
@@ -341,6 +341,36 @@ and view change histories.
 
 [The Google code review developer guide](https://google.github.io/eng-practices/review/) provides a useful set of guidelines for both reviewers and code authors.
 
+Below is a flowchart for the FIMS code review process. The author starts by submitting 
+a pull request (PR), ensuring documentation, tests, and CI checks are complete, 
+then propose a reviewer. The reviewer receives the review request and either 
+executes the review independently or pairs with another team representative 
+if assistance is needed. Based on the review, changes may be requested, which
+the author must address before approval. Once the PR is approved, the author 
+merges it into the main branch.
+
+```{r, echo=FALSE}
+DiagrammeR::mermaid("
+        graph TB
+        A[Author: Submit a pull request PR<br/><br/>- Documentation included<br/>- Tests implemented<br/>- CI checks passed]
+        B[Author: Propose a reviewer<br/><br/>- Default assignee: an OST rep for a quick review]
+        
+        A-->B
+        B-->C[Reviewer: Receive code review request]
+        C --> |No assistance needed| E[Reviewer: Execute code review]
+        C --> |Help wanted| D[Reviewer: Pair an OST rep with a center rep] 
```

Comment:
I do not think that we need to be so specific with "Pair an OST rep with a center rep" I think we can just say that the reviewer can add an additional reviewer to the PR.

Reply:
Done.